plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -184,9 +186,13 @@ public CompletableFuture<ConnectorSplitBatch> getNextBatch(ConnectorPartitionHan
 
         Iterator<FileScanTask> fileScanTasks = Iterators.limit(fileScanTaskIterator, maxSize);
         ImmutableList.Builder<ConnectorSplit> splits = ImmutableList.builder();
+        boolean skippedFiles = false;
```

Comment:
I think rather than keeping track of skippedFiles like this you could check the table handle's enforced constraint. If the constraint isAll, remove the equality deletes, otherwise leave them in?